### PR TITLE
Allow to configure sampling config for logs

### PR DESCRIPTION
**Pull request by bogdandrutu**
Fixes https://github.com/open-telemetry/opentelemetry-collector/issues/4554

### CODE REVIEW

File: service/telemetry/config.go [hunk 1]
```diff
@@ -91,6 +94,14 @@ type LogsConfig struct {
 	InitialFields map[string]interface{} `mapstructure:"initial_fields"`
 }
 
+// LogsSamplingConfig sets a sampling strategy for the logger. Sampling caps the
+// global CPU and I/O load that logging puts on your process while attempting
+// to preserve a representative subset of your logs.
+type LogsSamplingConfig struct {
```
Comment (codeboten) [inline service/telemetry/config.go:100]: Is it worth adding a note in the troubleshooting docs as to how to enable sampling? https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/troubleshooting.md#logs

**Review by codeboten — APPROVED** [review]
Change looks good, please rebase. Added just one non-blocking question

File: service/telemetry/config.go (hunk 1 above)
Comment (bogdandrutu) [reply service/telemetry/config.go:100]: We have only minimal examples there, not sure if we should add all of them there, or keep it minimal only the level.

Comment (bogdandrutu) [reply service/telemetry/config.go:100]: We can do it in a separate PR.

**Pull request merged**